src/histogram_io.jl

```diff
@@ -0,0 +1,82 @@
+_range_to_nt(r::AbstractRange) = (
+    start = first(r),
+    stop = last(r),
+    step = step(r),
+    length = length(r)
```

Comment:
We shouldn't need length, having `start`, `stop` and `step` should be sufficient, right?

Maybe `first` and `last` would be even better than `start` and `stop`, it's more semantic and less "mechanistic".

Reply:
Regarding length. We could drop it. But then we would have the same names as for a StepRange. There would be still a difference in the order (:start, :stop, :step) vs (:start, :step, :stop). Or we store also the Name as a symbol and use Val{:name} to dispatch for the correct type in the backconversion?

Reply:
I don't think we need to reconstruct the exact Julia type. Maybe it would be best to use `first:last` and `first:step:last` when reading.

Reply:
If we don't care about the exact Julia type we could just always use `StepRenLen` and store
just `:first`, `:stop`, `:step`?

Reply:
Not the exact type, but I think we should still make a difference between a range with and without `step`. But `first:last` and `first:step:last` should be good enough for that.

Reply:
Alright, then we are fine now I guess.